### PR TITLE
Update implementation_comparison.md

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/grpc/grpc-dotnet/issues/2188

The comparison page was last updated two years ago. This PR brings it up to date with features added since then.

### CODE REVIEW

File: doc/implementation_comparison.md [hunk 1]
```diff
@@ -22,12 +22,13 @@ Here are some key points in which the two implementation differ:
...
 - features available (see breakdown below)
+
 ## Frameworks supported
 
-Grpc.Core supports a wide range of .NET Framework versions, included some very old ones. A more detailed overview is [here]( https://github.com/grpc/grpc/tree/master/src/csharp#supported-platforms)
+Grpc.Core supports a wide range of .NET versions, including some very old ones. A more detailed overview is [here]( https://github.com/grpc/grpc/tree/master/src/csharp#supported-platforms)
 
-grpc-dotnet requires a more modern .NET Framework: A detailed summary of .NET Framework versions supported by grpc-dotnet is [here](https://docs.microsoft.com/en-us/aspnet/core/grpc/supported-platforms). Note that there's ongoing work to add [grpc-dotnet client support on
-legacy .NET Framework](https://docs.microsoft.com/en-us/aspnet/core/grpc/netstandard).
+grpc-dotnet is design for modern .NET versions. A detailed summary of .NET versions supported by grpc-dotnet is [here](https://docs.microsoft.com/aspnet/core/grpc/supported-platforms). There is limited support for [grpc-dotnet client support on
```
Comment (jtattermusch) [inline doc/implementation_comparison.md:30]: nit: "grpc-dotnet is design for" doesn't read well?

File: doc/implementation_comparison.md [hunk 2]
```diff
@@ -139,7 +144,7 @@ https://github.com/grpc/grpc/blob/master/doc/compression_cookbook.md
 
 Grpc.Core: supported (exposed publicly on Channel), provided by C-Core
 
-grpc-dotnet: not supported (to provide support, changes to .NET HttpClient are needed, so adding support is non-trivial).
+grpc-dotnet: supported
```
Comment (jtattermusch) [inline doc/implementation_comparison.md:147]: nit: does this require .NET5+ or .NET6+?  Might be worth documenting here. Same for wait-for-ready.

File: doc/implementation_comparison.md [hunk 3]
```diff
@@ -213,4 +218,22 @@ grpc-dotnet supports integration with `HttpClientFactory` via the Grpc.Net.Clien
...
+- URL parameter binding
+- JSON requests/responses
+
+gRPC can still be used to call services.
+
+### Addon: gRPC-Web
+
+grpc-dotnet supports the [gRPC-Web protocol](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md).
+
+gRPC-Web allows browser JavaScript and Blazor apps to call gRPC services. It's not possible to call an HTTP/2 gRPC service from a browser-based app. gRPC services hosted in ASP.NET Core can be configured to support gRPC-Web alongside HTTP/2 gRPC.
```
Comment (jtattermusch) [inline doc/implementation_comparison.md:237]: nit: "HTTP/2 gRPC" (multiple occurrences): we usually say "gRPC over HTTP/2"

**Review by jtattermusch — APPROVED** [review]
Thanks for the update. LGTM after addressing the comments.

File: doc/implementation_comparison.md (hunk 1 above)
Comment (JamesNK) [inline doc/implementation_comparison.md:30]:
```suggestion
grpc-dotnet uses features only available in modern .NET releases. It doesn't support some older versions of .NET. A detailed summary of .NET versions supported by grpc-dotnet is [here](https://docs.microsoft.com/aspnet/core/grpc/supported-platforms). There is limited support for [grpc-dotnet client support on
```

File: doc/implementation_comparison.md (hunk 2 above)
Comment (JamesNK) [reply doc/implementation_comparison.md:147]:
```suggestion
grpc-dotnet: Supported on .NET 5 or later
```

File: doc/implementation_comparison.md [hunk 4]
```diff
@@ -163,15 +168,15 @@ https://github.com/grpc/grpc/blob/master/doc/keepalive.md
 
 Grpc.Core: supported, provided by C-core
 
-grpc-dotnet: not supported (implementing the wait_for_ready flag requires supporting channel connectivity first).
+grpc-dotnet: supported
```
Comment (JamesNK) [inline doc/implementation_comparison.md:171]:
```suggestion
grpc-dotnet: Supported on .NET 5 or later
```

File: doc/implementation_comparison.md [hunk 5]
```diff
@@ -163,15 +168,15 @@ https://github.com/grpc/grpc/blob/master/doc/keepalive.md
...
+grpc-dotnet: Supported on .NET 5 or later
 
 https://github.com/grpc/grpc/blob/master/doc/wait-for-ready.md
 
 ### Naming / Resolver API
 
 Grpc.Core: conforms with the spec thanks to the C-core dependency. It is currently not possible to provide custom C# resolvers via resolver API (the APIs are in C core and aren't exposed in the C# layer).
 
-grpc-dotnet: only resolves basic DNS names. No API to provide a custom resolver.
+grpc-dotnet: Resolving is fully supported, and also provides an [API to write custom resolvers](https://learn.microsoft.com/aspnet/core/grpc/loadbalancingwrite-custom-resolvers-and-load-balancers).
```
Comment (JamesNK) [inline doc/implementation_comparison.md:179]:
```suggestion
grpc-dotnet: Resolving is fully supported on .NET 5 or later. grpc-dotnet also provides an [API to write custom resolvers](https://learn.microsoft.com/aspnet/core/grpc/loadbalancingwrite-custom-resolvers-and-load-balancers).
```

File: doc/implementation_comparison.md [hunk 6]
```diff
@@ -90,23 +93,25 @@ Both implementations fully support gRPC over HTTP/2 in both TLS and plaintext.
 
 Grpc.Core gets support for other transports supported by C-Core for free. Some (minor) integration work might be required to actually use these transports with gRPC C#.
 
-grpc-dotnet only supports the default transport.
+grpc-dotnet [supports interprocess communication (IPC)](https://learn.microsoft.com/aspnet/core/grpc/interprocess) using named pipes and Unix domain sockets.
```
Comment (JamesNK) [inline doc/implementation_comparison.md:96]:
```suggestion
grpc-dotnet supports other transports in .NET 5 or later. For example, grpc-dotnet [supports interprocess communication (IPC)](https://learn.microsoft.com/aspnet/core/grpc/interprocess) using named pipes and Unix domain sockets.
```

File: doc/implementation_comparison.md (hunk 3 above)
Comment (JamesNK) [inline doc/implementation_comparison.md:237]:
```suggestion
gRPC-Web allows browser JavaScript and Blazor apps to call gRPC services. It's not possible to call a gRPC service over HTTP/2 from a browser-based app. gRPC services hosted in ASP.NET Core can be configured to support gRPC-Web alongside gRPC over HTTP/2.
```

**Pull request merged**